demux/packet.c

```diff
@@ -49,9 +50,18 @@ static void packet_destroy(void *ptr)
     demux_packet_unref_contents(dp);
 }
 
-static struct demux_packet *packet_create(void)
+static struct demux_packet *packet_create(struct demux_packet_pool *pool)
 {
-    struct demux_packet *dp = talloc(NULL, struct demux_packet);
+    struct demux_packet *dp = pool ? demux_packet_pool_pop(pool) : NULL;
+    struct AVPacket *avpkt = NULL;
+    if (dp) {
+        avpkt = dp->avpacket;
+        if (avpkt)
+            av_packet_unref(avpkt);
```

Comment:
wait, shouldn't this happen before stuff gets put into the pool?

Reply:
Then I have to ask because I'm not sure: is it possible that an `AVPacket` references and keeps hardware resources alive if we don't unref it?

Reply:
The whole thing is intended for demux packets, there are not hardware resource tied to that. If they were we couldn't cache anything, which we do currently. This commit just extends the lifetime of the cached packets, there is not behavior change in principle.